From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rmp22 <195054967+rmp22@users.noreply.github.com>
Date: Fri, 1 May 2026 06:36:10 +0800
Subject: [PATCH 167/226] services: Disabling debug pss calls

expensive especially now that we added recents memory info - neglible impact on calculation since the info is usual N x 1024L added to mem info compuatation

Change-Id: I03b4fe9ddd416819b5806cdc66fb81fe4f384722
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 .../java/com/android/server/am/ActivityManagerService.java | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 14e3675237..90824cde09 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -4117,6 +4117,12 @@ public class ActivityManagerService extends IActivityManager.Stub
     public long[] getProcessPss(int[] pids) {
         enforceNotIsolatedCaller("getProcessPss");
 
+        final long[] pss = new long[pids.length];
+
+        if (!Build.IS_ENG) {
+            return pss;
+        }
+
         final int callingPid = Binder.getCallingPid();
         final int callingUid = Binder.getCallingUid();
         final int userId = UserHandle.getUserId(callingUid);
@@ -4126,7 +4132,6 @@ public class ActivityManagerService extends IActivityManager.Stub
         final boolean allUids = mAtmInternal.isGetTasksAllowed(
                 "getProcessPss", callingPid, callingUid);
 
-        final long[] pss = new long[pids.length];
         for (int i=pids.length-1; i>=0; i--) {
             ProcessRecord proc;
             int oomAdj;
